### PR TITLE
Minor Corrections In Thrift Docs

### PR DESCRIPTION
**Pull request by saiteja13427**
Corrections to minor grammatical mistakes and typos.

**Note:** I was just going through several repositories at meta to look for feasible contribution opportunities and in that process found many small mistakes in the docs. So, thought to start my contribution with this itself. 

Hope it helps perfect the repo!

### CODE REVIEW

File: thrift/doc/Cpp2.md [hunk 1]
```diff
@@ -37,9 +37,9 @@ similar to the older first-generation implementation:
   to prevent large allocations of memory.
 
 * Allows true zero-copy from client to server and back using IOBufs
-  (vs. previous generated code that used std::string for binary type).
+  (vs. previously generated code that used std::string for binary type).
```
Comment (yfeldblum) [inline thrift/doc/Cpp2.md:40]: Existing is correct; revert.

File: thrift/doc/Cpp2.md [hunk 2]
```diff
@@ -105,10 +105,10 @@ Will generate an interface similar to
       ...
     }
 
-So you have three choices of handler type to implement:
+So you have three choices of handler types to implement:
```
Comment (yfeldblum) [inline thrift/doc/Cpp2.md:108]: Existing is correct; revert.

File: thrift/doc/mstch.md [hunk 3]
```diff
@@ -132,7 +132,7 @@ about the context.
 
 Lists represent zero or more elements of some other type, and function similar
 to for-each loops in other languages. The body of the loop will be expanded
-once per element in the list, and on each iteration the context will be
+once per element in the list, and on each iteration, the context will be
```
Comment (yfeldblum) [inline thrift/doc/mstch.md:135]: Existing is better; revert.

Alternatively,
```suggestion
once per element in the list, and, on each iteration, the context will be
```

File: thrift/doc/mstch.md [hunk 4]
```diff
@@ -327,7 +327,7 @@ mstch::map extend_struct(const t_struct& strct) const override {
 ```
 
 Where `my_escaping_function` is some function that performs the desired
-escaping. From within templates now, as long as a struct is currently in scope,
+escape. From within templates now, as long as a struct is currently in scope,
```
Comment (yfeldblum) [inline thrift/doc/mstch.md:330]: Existing is correct; revert.

**Review by yfeldblum — CHANGES_REQUESTED** [review]
Thanks for the suggestions. Back to you for tweaks.

File: thrift/doc/Cpp2.md (hunk 1 above)
Comment (saiteja13427) [reply thrift/doc/Cpp2.md]: Reverted

File: thrift/doc/Cpp2.md (hunk 2 above)
Comment (saiteja13427) [reply thrift/doc/Cpp2.md]: Reverted

File: thrift/doc/mstch.md (hunk 3 above)
Comment (saiteja13427) [reply thrift/doc/mstch.md]: Added alternative

File: thrift/doc/mstch.md (hunk 4 above)
Comment (saiteja13427) [reply thrift/doc/mstch.md]: Reverted

**Comment by vitaut** [comment]
Please rebase the PR as some of the files have moved.

**Comment by saiteja13427** [comment]
@vitaut Did the rebase.

Please review and merge!

**Comment by vitaut** [comment]
Merged, thanks!

**Pull request closed without merging**